lib/icinga/externalcommandprocessor.cpp

```diff
@@ -970,10 +971,19 @@ void ExternalCommandProcessor::ScheduleSvcDowntime(double, const std::vector<Str
 void ExternalCommandProcessor::DelSvcDowntime(double, const std::vector<String>& arguments)
 {
 	int id = Convert::ToLong(arguments[0]);
-	Log(LogNotice, "ExternalCommandProcessor")
-		<< "Removing downtime ID " << arguments[0];
+	m_DownTimeRemoved = false;
 	String rid = Downtime::GetDowntimeIDFromLegacyID(id);
-	Downtime::RemoveDowntime(rid, true);
+
+	try {
+		Downtime::RemoveDowntime(rid, true);
+	} catch (std::runtime_error& error) {
+		m_DownTimeRemoved = true;
+		Log(LogWarning, error.what());
+	}
+
+	if (!m_DownTimeRemoved)
+		Log(LogNotice, "ExternalCommandProcessor")
+				<< "Removing downtime ID " << arguments[0];
```

Comment:
Or much simpler, you could just move this message at the end within the try block. Then you shouldn't need `m_DownTimeRemoved` at all.

Also as the message is now printed after removing succeeded, I'd change it to "Removed".